File: src/generateChatData/formObjects.py
class answer:

    def __init__(self, value, qType, name):
        self.cf_label = value
        self.value = value
        self.type = qType
        self.tag = "input"
        self.name = name

    def condition(self, question, answer):
        setattr(self, "cf_conditional_" + question, answer)

    def addName(self, name):
        self.name = name

class question: 

    def __init__(self, name, question, placeholder, qType, answers = None, fieldset = True):
        self.name = name
        self.cf_questions = question
        self.cf_input_placeholder = placeholder
        self.type = qType
        if fieldset:
            self.tag = "fieldset"
        else :
            self.tag = "input"
        if answers is not None:
            self.children = []
            for a in answers:
                if qType == "Checkboxes":
                    r = answer(a, "checkbox", name)
                else :
                    r = answer(a, "radio", name)
                # r.addName(type(self).__name__)
                self.children.append(r)

    def addChild(self, c):
        self.children.append(c)
        c.addName(type(self).__name__)

    def conditionOn(self, question, cond):
        if self.tag == "fieldset":
            for c in self.children:
                c.condition(question, cond)
        else :
            self.condition(question, cond)

    def condition(self, question, cond):
        setattr(self, "cf_conditional_" + question, cond)


class robotMessage:
    
  def __init__(self, message):
    self.cf_questions = message
    self.tag = "cf-robot-message"
  
  def conditionOn(self, question, cond):
    setattr(self, "cf_conditional_" + question, cond)

File: src/generateChatData/formObjects.js
class answer {
    constructor(value, qType, name) {
        this.cf_label = value;
        this.value = value;
        this.type = qType;
        this.tag = "input";
        this.name = name;
    }
    condition(question, answer) { // not working
        this["cf_conditional_" + question] = answer
    }
    addName(name){
        this.name = name;
    }
  }

class question {
    constructor(name, question, placeholder, qType, answers = null, fieldset = true){
        this.name = name;
        this.cf_questions = question;
        this.cf_input_placeholder = placeholder;
        this.type = qType;
        if(fieldset) {
            this.tag = "fieldset"
        } else {
            this.tag = "input"
        }
        if(answers !== null) {
            this.children = [];
            var a
            for(a of answers) {
                var r
                if(qType == "Checkboxes") {
                    r = new answer(a, "checkbox", name)
                } else {
                    r = new answer(a, "radio", name)
                }
                this.children.push(r)
            }
        }
    }

    addChild(c){
        this.children.push(c)
        c.addName(this.name)
    }

    conditionOn(question, cond) {
        if(this.tag == "fieldset") {
            var c
            for (c of this.children) {
                c.condition(question, cond)
            }
        } else {
            this.condition(question, cond)
        }
    }

    condition(question, cond) {
        this["cf_conditional_" + question] = answer
    }

}

class robotMessage {
    constructor(message){
        this.questions = message;
        this.tag = "cf-robot-message"
    }
    condition(question, cond) {
        this["cf_conditional_" + question] = answer
    }
}

// myanswer = new answer("Ford", "bla", "name");
// myanswer.addName("2ndName");
// myanswer.condition("cond", "ans")
// myquestion = new question("first", "q", "ph", "text", ["answerA", "answerB"]);

// console.log(myanswer);
// console.log(myquestion);